### PR TITLE
accounts/abi: fix panic in MethodById lookup. Fixes #17797

### LINKED ISSUE #17797: ABI MethodById panic
when parse #4999272 block, i use erc20 abi get method panic, transaction hash is : `0xbefd2f3a0288731f4db3e24995a1e45c68403a3991e64f00f62b800c22e778c4`

```
m, err := ERC20ABI.MethodById(tx.Data())
```

### PR DESCRIPTION
**Pull request by holiman**
Trivial fix,

### CODE REVIEW

File: accounts/abi/abi_test.go [hunk 1]
```diff
@@ -711,5 +711,18 @@ func TestABI_MethodById(t *testing.T) {
...
+	}
+	if _, err := abi.MethodById([]byte{}); err == nil {
+		t.Errorf("Expected error, too short to decode data")
+	}
+	if _, err := abi.MethodById(nil); err == nil {
+		t.Errorf("Expected error, nil is short to decode data")
+	}
+}
+
+func TestABI_MethodByIdEmpty(t *testing.T) {
```
Comment (gballet) [inline accounts/abi/abi_test.go]: I guess this is a leftover from a previous test, and that you moved the test to `TestABI_MethodById` ? Anyhow, no need to commit an empty function.

File: accounts/abi/abi.go [hunk 2]
```diff
@@ -137,6 +137,9 @@ func (abi *ABI) UnmarshalJSON(data []byte) error {
 // MethodById looks up a method by the 4-byte id
 // returns nil if none found
 func (abi *ABI) MethodById(sigdata []byte) (*Method, error) {
+	if len(sigdata) < 4 {
```
[the remark this answers is not in the document]
Comment (holiman) [reply accounts/abi/abi.go]: Ahmm.. really? `MethodById looks up a method by the 4-byte id` -- it's a 4-byte id, do we really have to make it generic for N-byte ids?

[the remark this answers is not in the document]
Comment (gballet) [reply accounts/abi/abi.go]: it's not about being generic (a variable) but documenting (a constant name). Alright, it's not a blocker let's move on.

**Comment by holiman** [comment]
Fixed, squashed

**Pull request merged**